components/Footer/LanguageSwitcher.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { motion } from 'framer-motion';
+import { useTranslation } from 'next-i18next';
+import { useRouter } from 'next/router';
+import rusFlag from '@/public/images/LanguageSwitcher/rus.svg';
+import ukFlag from '@/public/images/LanguageSwitcher/uk.svg';
+import styles from './LanguageSwitcher.module.css';
+
+const LanguageSwitcher = () => {
+  const { i18n } = useTranslation();
+  const router = useRouter();
+  const [language, setLanguage] = useState(i18n.language);
+
+  const toggleLanguage = () => {
+    const scrollPosition = window.scrollY;
+
+    const newLanguage = language === 'ru' ? 'en' : 'ru';
+    setLanguage(newLanguage);
+
+    i18n.changeLanguage(newLanguage).then(() => {
+      router
+        .push(router.pathname, router.asPath, { locale: newLanguage })
+        .then(() => {
+          window.scrollTo(0, scrollPosition);
+        });
+    });
+  };
+
+  const isRussian = language === 'ru';
+  const currentFlag = isRussian ? rusFlag : ukFlag;
+
+  const gradients = {
+    russian:
+      'linear-gradient(15deg, rgba(255, 0, 0, 0.3) , rgba(38, 128, 212, 0.3),rgba(255, 255, 255, 0.3))',
+    english:
+      'linear-gradient(-15deg, rgba(38, 128, 212, 0.3), rgba(255, 255, 255, 0.3), rgba(255, 0, 0, 0.3))',
+  };
+
```

Comment:
Можно упростить.
В глобальных константах будет 
export const APP_LANGUAGE = {
  RU: 'ru',
  EN: 'en',
};

```suggestion
  const isRussian = language === 'ru';
  const currentFlag = isRussian ? rusFlag : ukFlag;

  const gradients = {
    APP_LANGUAGE.RU:
      'linear-gradient(15deg, rgba(255, 0, 0, 0.3) , rgba(38, 128, 212, 0.3),rgba(255, 255, 255, 0.3))',
    APP_LANGUAGE.EN:
      'linear-gradient(-15deg, rgba(38, 128, 212, 0.3), rgba(255, 255, 255, 0.3), rgba(255, 0, 0, 0.3))',
  };

```

И там, где нужен градиент, брать его как gradients[language] .